src/pretalx/submission/models/resource.py

```diff
@@ -65,3 +73,25 @@ def filename(self):
         with suppress(ValueError):
             if self.resource:
                 return Path(self.resource.name).name
+
+    @cached_property
+    def is_available(self):
+        """Check if the resource is available based on hide_until_event_day setting.
+
+        If hide_until_event_day is True, the resource is only available on or after
+        the start time of the submission's earliest scheduled slot.
+        """
+        if not self.hide_until_event_day:
+            return True
+
+        # Get the earliest slot start time for this submission
+        earliest_slot = (
+            self.submission.slots.filter(start__isnull=False).order_by("start").first()
+        )
+
+        if not earliest_slot or not earliest_slot.start:
+            # No scheduled slot yet, so resource is not available
+            return False
+
+        # Resource is available if current time is on or after the slot start
+        return now() >= earliest_slot.start
```

Comment:
The user is told that the resource will start being visible on the day of the session, but per this code, it only starts being visible at the *time* of the session.